src/GovUk.Education.ExploreEducationStatistics.Admin/Services/EmbedBlockService.cs

```diff
@@ -0,0 +1,153 @@
+using GovUk.Education.ExploreEducationStatistics.Admin.Validators;
+using GovUk.Education.ExploreEducationStatistics.Admin.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Release = GovUk.Education.ExploreEducationStatistics.Content.Model.Release;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services;
+
+public class EmbedBlockService : IEmbedBlockService
+{
+    private readonly ContentDbContext _contentDbContext;
+    private readonly IPersistenceHelper<ContentDbContext> _persistenceHelper;
+    private readonly IUserService _userService;
+    private readonly IMapper _mapper;
+
+    public EmbedBlockService(
+        ContentDbContext contentDbContext,
+        IPersistenceHelper<ContentDbContext> persistenceHelper,
+        IUserService userService,
+        IMapper mapper)
+    {
+        _contentDbContext = contentDbContext;
+        _persistenceHelper = persistenceHelper;
+        _userService = userService;
+        _mapper = mapper;
+    }
+
+    public async Task<Either<ActionResult, EmbedBlockLinkViewModel>> Create(Guid releaseId,
+        EmbedBlockCreateRequest request)
+    {
+        return await _persistenceHelper
+            .CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async _ =>
+            {
+                return await _persistenceHelper
+                    .CheckEntityExists<ContentSection>(request.ContentSectionId, query =>
+                        query.Include(contentSection => contentSection.Content))
+                    .OnSuccess<ActionResult, ContentSection, EmbedBlockLinkViewModel>(async contentSection =>
+                    {
+                        if (!_contentDbContext.ReleaseContentSections
+                              .Any(rcs => rcs.ReleaseId == releaseId
+                                          && rcs.ContentSectionId == request.ContentSectionId))
+                        {
+                            return ValidationUtils.ValidationActionResult(
+                                ValidationErrorMessages.ContentSectionNotAttachedToRelease);
+                        }
+
+                        var embedBlock = new EmbedBlock
+                        {
+                            Id = Guid.NewGuid(),
+                            Title = request.Title,
+                            Url = request.Url,
+                        };
+
+                        var order = contentSection.Content.Any()
+                            ? contentSection.Content.Max(contentBlock => contentBlock.Order) + 1
```

Comment:
Just wondering if we need to account for negative values here? i.e. if a user adds an order such as `-10`

Reply:
I think it's ok to leave as it is. I'm pretty sure other areas of the codebase have the same potential flaw. 

It will preserve the order even if there were negative numbers - but it seems unlikely that would happen. I don't think there is any security issue with this, so while it would be possible for someone to add content blocks with unexpected order numbers, it wouldn't cause any issues - at least I don't think.